plugins/synaptics-rmi/fu-synaptics-rmi-v7-device.c

```diff
@@ -123,6 +123,52 @@ fu_synaptics_rmi_v7_device_detach(FuDevice *device, FuProgress *progress, GError
 	return TRUE;
 }
 
+static gboolean
+fu_synaptics_rmi_v7_device_erase_partition(FuSynapticsRmiDevice *self,
+					   guint8 partition_id,
+					   GError **error)
+{
+	FuSynapticsRmiFunction *f34;
+	FuSynapticsRmiFlash *flash = fu_synaptics_rmi_device_get_flash(self);
+	g_autoptr(GByteArray) erase_cmd = g_byte_array_new();
+	/* f34 */
+	f34 = fu_synaptics_rmi_device_get_function(self, 0x34, error);
+	if (f34 == NULL)
+		return FALSE;
+	fu_byte_array_append_uint8(erase_cmd, partition_id);
+	fu_byte_array_append_uint32(erase_cmd, 0x0, G_LITTLE_ENDIAN);
+	fu_byte_array_append_uint8(erase_cmd, RMI_FLASH_CMD_ERASE);
+
+	fu_byte_array_append_uint8(erase_cmd, flash->bootloader_id[0]);
+	fu_byte_array_append_uint8(erase_cmd, flash->bootloader_id[1]);
+
+	g_usleep(1000 * 1000);
```

Comment:
a delayis needed  _before_ the write?

Reply:
Yes it's needed in new BL.
It isn't needed in older BL, but it won't cause the problem besides of time.